pkg/sql/show_tenant.go

```diff
@@ -12,33 +12,87 @@ package sql
 
 import (
 	"context"
+	"time"
 
+	"github.com/cockroachdb/cockroach/pkg/repstream/streampb"
 	"github.com/cockroachdb/cockroach/pkg/roachpb"
 	"github.com/cockroachdb/cockroach/pkg/sql/catalog/colinfo"
 	"github.com/cockroachdb/cockroach/pkg/sql/catalog/descpb"
 	"github.com/cockroachdb/cockroach/pkg/sql/sem/tree"
+	"github.com/cockroachdb/cockroach/pkg/util/hlc"
+	"github.com/cockroachdb/cockroach/pkg/util/log"
+	"github.com/cockroachdb/cockroach/pkg/util/timeutil"
+	"github.com/cockroachdb/errors"
 )
 
 type showTenantNode struct {
-	columns colinfo.ResultColumns
-	name    roachpb.TenantName
-	info    *descpb.TenantInfo
-	done    bool
+	name               tree.Expr
+	info               *descpb.TenantInfo
+	withReplication    bool
+	replicationInfo    *streampb.StreamIngestionStats
+	protectedTimestamp hlc.Timestamp
+	columns            colinfo.ResultColumns
+	done               bool
 }
 
 func (p *planner) ShowTenant(_ context.Context, n *tree.ShowTenant) (planNode, error) {
-	return &showTenantNode{
-		name:    roachpb.TenantName(n.Name),
-		columns: colinfo.TenantColumns,
-	}, nil
+	node := &showTenantNode{
+		name:            n.Name,
+		withReplication: n.WithReplication,
+	}
+	if n.WithReplication {
+		node.columns = colinfo.TenantColumnsWithReplication
+	} else {
+		node.columns = colinfo.TenantColumns
+	}
+	return node, nil
 }
 
 func (n *showTenantNode) startExec(params runParams) error {
-	info, err := params.p.GetTenantInfo(params.ctx, n.name)
+	if err := params.p.RequireAdminRole(params.ctx, "show tenant"); err != nil {
+		return err
+	}
+
+	if err := rejectIfCantCoordinateMultiTenancy(params.p.execCfg.Codec, "show"); err != nil {
+		return err
+	}
+
+	info, err := GetTenantRecordByName(params.ctx, params.p.execCfg, params.p.Txn(), roachpb.TenantName(n.name.String()))
 	if err != nil {
 		return err
 	}
 	n.info = info
+	if n.withReplication {
+		if info.TenantReplicationJobID == 0 {
+			return errors.Newf("tenant %q does not have an active replication job", n.name)
+		}
+		mgr, err := params.p.EvalContext().StreamManagerFactory.GetStreamIngestManager(params.ctx)
+		if err != nil {
+			return err
+		}
+		stats, err := mgr.GetStreamIngestionStats(params.ctx, info.TenantReplicationJobID)
+		// An error means we don't have stats but we can still present some info,
```

Comment:
good question, we have most stats but not all because we fail here:
https://github.com/cockroachdb/cockroach/blob/721cc761de83e1424de2ec377c6078983ea11e4f/pkg/ccl/streamingccl/streamingest/stream_ingestion_job.go#L106

I think it will be nice to refactor and return what we can.

Reply:
done.